mis_builder/models/mis_report_instance.py

```diff
@@ -431,6 +433,30 @@ def _default_company_id(self):
     date_from = fields.Date(string="From")
     date_to = fields.Date(string="To")
     temporary = fields.Boolean(default=False)
+    mis_report_view = fields.Text(string='MIS Report view',
+                                  compute='_compute_mis_report_view')
+
+    def _compute_mis_report_view(self):
+        """This method is dummy because we use this field as a placeholder
+        to render the actual mis report instance in the html format using the
+        widget"""
+        for rec in self:
+            rec.mis_report_view = ""
+
+    @api.model
+    def get_mis_report_view_html(self):
+        mis_report_data = self.compute()
+        rcontext = {}
+        rcontext['mis_report_data'] = mis_report_data
+        rcontext['mis_report_data']['report_name'] = self.name
+        rcontext['mis_instance_id'] = self.id
+        return self.env.ref(
+            'mis_builder.MisReportInstance').render(rcontext)
+
+    @api.multi
+    def get_mis_report_view_from_id(self):
+        for rec in self:
+            return rec.get_mis_report_view_html()
```

Comment:
This return inside a for loop is suspicious. If this method is really necessary, there should be an ensure_one and no loop. But then it does the same as get_mis_report_view_html so it's probably useless?